### DISCUSSION TITLE
`Generate a Commit Message` maybe should consider the contents of the git index

### DISCUSSION THREAD

**Discussion by gh-liu** [olimorris/codecompanion.nvim#273] [General]
As the doc of `git-commit` says: `Create a new commit containing the current contents of the index and the given log message describing the changes.`, maybe `Generate a Commit Message` should consider the contents of the git index .
https://github.com/olimorris/codecompanion.nvim/blob/4bec50da26b411e52accdabd358e7c00ff94d2d3/lua/codecompanion/config.lua#L645

change `git diff` to `git diff --staged`.

**Comment by dhruvinsh** [comment]
there is same discuss exists, https://github.com/olimorris/codecompanion.nvim/discussions/132#discussioncomment-10778832

Also with v5.0.0 it is by default set to `git diff --staged`
